src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -252,7 +253,8 @@ class UserService(
       }
     }
 
-    deliusUser.teams?.firstOrNull()?.let { team ->
+    val userTeams = orderUserDeliusTeams(deliusUser)
+    userTeams?.firstOrNull()?.let { team ->
```

Comment:
we call orderUserDetailsTeam(deliusUser) twice in this function. Could call it once and capture the firstOrNull in a variable e.g.

val firstUserTeam = orderUserDeliusTeams(deliusUser)?.firstOrNull()

This of course slightly changes the approach suggested in my previous comment!

Reply:
We calling it once only in this function

Reply:
i can see orderUserDeliusTeams being used on 256 and 222 but this is just minor duplication so happy to merge if you're happy

Reply:
I changed the code by removing the function findProbationDeliveryUnit and get the last Borough code for the user as you suggested